blood overlay handling rework

CakeQ opened the pull request:
🆑 Cakey
tweak: Filth overlays like blood and oil now tracks a transferral value for each item. Items with filth transferral will transfer their blood/oil to whatever they interact with.
tweak: interacting with items with bloody hands or gloves will transfer that blood/oil to the item depending on how filthy you are.
tweak: Squashed some legacy values related to blood overlays for items like shoes and items in general
tweak: Forensic DNA from blood will now transfer to items when interacting with them with bloody hands/gloves
tweak: Hugging people or grabbing them while covered in blood/oil will transfer it.
bugfix: Fixed coat and armor layered filth overlays using incorrect icon states
/🆑

Tried to modernise blood overlay code a bit by adding more custom proccalls for applying blood that isn't directly from a human. All other code routes through these custom procs. It's still all named after blood rather than a generic filth system because I couldn't be bothered

Also added some new procs that can be called to transfer filth from items directly to other items.

![image](https://github.com/user-attachments/assets/d4fb024a-ad4e-4a42-b39e-409b86c3a67b)

SierraKomodo reviewed (dismissed):
A couple things noted from brief testing:
- Stabbing a person with a knife did not apply any blood overlays to the knife itself.
- Blood overlays on my hand were white instead of read.
- Walking through the resulting blood spatters doesn't appear to be creating footprints or bloodying feet.

CakeQ commented:
Fixed attacks not bloodying the weapon. I moved some DNA tracking further up the chain that prevented the initial overlay from being applied. Didn't catch it as transfering blood to items you hold in your bloodied hands was working as intended through a new function.

Not having the issue with blood being white on my end, what mob/decal type were you using to test? I tested with monkeys & humans. 

walking through blood is fine as well for me. I'd note that the singular drips have an amount of 0 transfer units (I haven't changed this)

![image](https://github.com/user-attachments/assets/460e716f-1544-4d37-9229-5efe2bdf6ad7)

SierraKomodo reviewed (changes requested):
Did another round of testing:
- Turfs that have existing blood splatters (`/obj/decal/cleanable/blood/splatter`) aren't creating footprint trails anymore.
- If you're barefoot, there's no blood transfer or footprint trails generated at all.
- For the white blood overlays on hands, noticed it's happening to the person being stabbed too. In both cases, no gloves. Both mobs were base humans. Decal no idea, my testing was just stab with a knife until bloodied.

CakeQ commented:
Yeah not gonna lie I'm pretty blind for not seeing the white hands in my own screenshot

CakeQ commented:
Issues fixed, decal blood transferral for bare feet was gated behind a shoes nullcheck that it shouldn't have been. Also fixed white hands (wasn't applying to hands_blood_color)

CakeQ committed:
blood overlay handling rework

SierraKomodo reviewed (dismissed):
Rebase

CakeQ commented:
rebased
